### ISSUE TITLE
Dandelion Implementation / Spec

### ISSUE THREAD

**Issue by kaiserd** [ChainSafe/js-libp2p-gossipsub#341]
## Background

As I am working on a Dandelion [Vac RFC](https://rfc.vac.dev/) specifying Dandelion for [Waku Relay](https://rfc.vac.dev/spec/11/)
(and by extension for libp2p, because Waku Relay builds on libp2p gossipsub),
I was looking into current implementations and analyses beyond the Dandelion/Dandelion++ papers.
The first goal is identifying anonymity properties,
and, if possible, relaxations that yield lower latency but still beneficial anonymity properties.

## js-libp2p-gossipsub Dandelion Implementation

The [current implementation](https://github.com/ChainSafe/js-libp2p-gossipsub/blob/dandelion/publish/ts/index.ts) of js-libp2p-gossipsub does not follow the Dandelion and/or Dandelion++ approaches directly.
It currently implements what the Dandelion paper refers to as "diffusion-by-proxy" with a fixed stem length
(the implementation features an additional degree parameter for the anonymity phase, though).

In contrast, Dandelion

1) fixes the anonymity graph
2) has a probabilistic stem length
3) introduces random delay in the fluff phase (because it uses diffusion) which gossipsub does not do

Also, it introduces epochs and a fail-safe that diffuses messages if they are not received via fluff spreading after some time.

I wanted to ask if there is any analysis regarding the design choices made in the current implementation?
It will improve latency, but (presumably) at the cost of lower entropy (i.e. higher recall / precision for the attacker).

## Dandelion Vac Spec

Would any js-libp2p-gossipsub Dandelion contributor be interested in collaborating on a [Vac RFC](https://rfc.vac.dev/) specifying Dandelion for libp2p gossipsub / [Waku Relay](https://rfc.vac.dev/spec/11/)? I am currently working on a first raw version as a basis for discussion. We are interested in trade-offs leaning towards lower latency at a low anonymity cost.

## Further References

* [Ethereum consensus layer validator anonymity using Dandelion++ and RLN conclusion](https://ethresear.ch/t/ethereum-consensus-layer-validator-anonymity-using-dandelion-and-rln-conclusion/12698).
* [On the Anonymity of Peer-To-Peer Network Anonymity Schemes Used by Cryptocurrencies](https://arxiv.org/pdf/2201.11860.pdf)

**Comment by dapplion** [comment]
CC: @noot @wemeetagain are you avail to participate in the spec-ing process?

**Comment by AtHeartEngineer** [comment]
> I'd be interested in potentially collaborating on this. would also be good to loop in @AtHeartEngineer since he's actively researching this as well.

Yep I have been talking to @kaiserd some too. I'm going to figure out a good place for us all to talk/coordinate.
